src/decopatch/main.py

```diff
@@ -1,21 +1,63 @@
-from makefun import with_signature, add_signature_parameters
+from makefun import add_signature_parameters, with_signature
+
+from decopatch.utils_calls import (call_in_appropriate_mode,
+                                   no_parenthesis_usage,
+                                   with_parenthesis_usage)
+from decopatch.utils_disambiguation import (
+    DecoratorUsageInfo, FirstArgDisambiguation, can_arg_be_a_decorator_target,
+    create_single_arg_callable_or_class_disambiguator, disambiguate_call)
 from decopatch.utils_modes import SignatureInfo, make_decorator_spec
-from decopatch.utils_disambiguation import create_single_arg_callable_or_class_disambiguator, disambiguate_call, \
-    DecoratorUsageInfo, can_arg_be_a_decorator_target
-from decopatch.utils_calls import with_parenthesis_usage, no_parenthesis_usage, call_in_appropriate_mode
 
 try:  # python 3.3+
-    from inspect import signature, Parameter
+    from inspect import Parameter, signature
 except ImportError:
-    from funcsigs import signature, Parameter
+    from funcsigs import Parameter, signature
 
 try:  # python 3.5+
-    from typing import Callable, Any, Optional
+    from typing import Any, Callable, Optional
```

Comment:
We can now probably get rid of this python 3.5 try/except

Reply:
Oh wait, actually not.

We're still supporting 2.7. Maybe better drop it?

pytest, for example, supports 3.6+